src/client/eppo-client.ts

```diff
@@ -693,16 +694,23 @@ export default class EppoClient {
       };
     } catch (error) {
       const eppoValue = this.rethrowIfNotGraceful(error, defaultValue);
-      const flagEvaluationDetails = new FlagEvaluationDetailsBuilder(
-        '',
-        [],
-        '',
-        '',
-      ).buildForNoneResult('ASSIGNMENT_ERROR', `Assignment Error: ${error.message}`);
-      return {
-        eppoValue,
-        flagEvaluationDetails,
-      };
+      if (error instanceof FlagEvaluationError && error.flagEvaluationDetails) {
```

Comment:
`EppoValue.valueOf()` on line 692 may throw an error that is not `FlagEvaluationError`. That will loose evaluation details

Reply:
You're right, it will. I made another update to fix that.